--- a/src/file_io.c
+++ b/src/file_io.c
@@ -192,6 +192,19 @@ int _1905_read_set_config(struct p1905_m
 #endif
 		hidden_ssid = 0;
 
+		/*skip commented out content*/
+		if (strncmp(pos1, "#", 1) == 0) {
+			/*go to next line*/
+			pos2 = strchr(pos1, '\n');
+			if (pos2 == NULL) {
+			debug(DEBUG_ERROR, "get next line fail\n");
+			goto err;
+		}
+			pos2++;
+			pos1 = pos2;
+			continue;
+		}
+
 		/*index*/
 		pos2 = strchr(pos1, ',');
 		if (pos2 == NULL) {
@@ -537,6 +550,14 @@ int _1905_read_set_config(struct p1905_m
 				} else {
 					debug(DEBUG_TRACE, "pvid not found or too far\n");
 				}
+				/*go to next line*/
+				pos2 = strchr(pos1, '\n');
+				if (pos2 == NULL) {
+					debug(DEBUG_ERROR, "get next line fail\n");
+					goto err;
+				}
+				pos2++;
+				pos1 = pos2;
 			}
 		}
 #endif
